sklearn/neighbors/base.py

```diff
@@ -276,9 +278,25 @@ def _pairwise(self):
 class KNeighborsMixin(object):
     """Mixin for k-neighbors searches"""
 
+    def _kneighbors_reduce_func(self, dist, start,
+                                n_neighbors, return_distance):
```

Comment:
Would it help readability to decorate it with `@staticmethod` and remove `self` to indicate that it doesn't interact with `self`?

Reply:
It uses `effective_metric_`